### PR TITLE
Change RandomState to seed value.

### PR DESCRIPTION
**Pull request by RothNRK**
Change RandomState to seed to avoid the unintended behaviour caused by calling random_train_test_split with the same random_state.

In addition to changing `random_state` to a simple `seed` value I changes the import path to explicit paths instead of depending on the package's `__all__`

Addresses issue: https://github.com/lyst/lightfm/issues/328

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 import scipy.sparse as sp
 
-from lightfm import LightFM
+from lightfm.lightfm import LightFM
```
Comment (maciejkula) [inline tests/test_api.py]: Thanks!

File: lightfm/cross_validation.py [hunk 2]
```diff
@@ -37,8 +37,8 @@ def random_train_test_split(interactions,
         The interactions to split.
     test_percentage: float, optional
         The fraction of interactions to place in the test set.
-    random_state: np.random.RandomState, optional
-        The random state used for the shuffle.
+    seed: int, optional
```
Comment (maciejkula) [inline lightfm/cross_validation.py]: I think the argument name should remain `random_state`, but the code should be changed to do the following:
1. If a `RandomState` instance is passed, behave exactly as before.
2. If something else is passed, construct a random state from it, treating it as a seed.

The reason is backward compatibility:
1. Uses in the wild will rely on the argument name.
2. Uses in the wild will rely on the function working with a `RandomState` instance.

An (slightly overcomplicated) example of this is [here](https://github.com/lyst/lightfm/blob/master/lightfm/lightfm.py#L201).

File: .pytest_cache/v/cache/lastfailed [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (maciejkula) [inline .pytest_cache/v/cache/lastfailed]: Do we need this file?

File: .pytest_cache/v/cache/nodeids [hunk 4]
```diff
@@ -0,0 +1,68 @@
+[
```
Comment (maciejkula) [inline .pytest_cache/v/cache/nodeids]: Do we need this file?

File: lightfm/cross_validation.py (hunk 2 above)
Comment (RothNRK) [reply lightfm/cross_validation.py]: That makes sense. I'll get that in there.

Comment (RothNRK) [reply lightfm/cross_validation.py]: I think the docs should be worded in a way that makes it clear that providing an integer as a seed is preferred over providing a np.RandomState. Is this ok with you or would you prefer to keep the language neutral?

File: lightfm/cross_validation.py [hunk 5]
```diff
@@ -37,8 +37,9 @@ def random_train_test_split(interactions,
         The interactions to split.
     test_percentage: float, optional
         The fraction of interactions to place in the test set.
-    random_state: np.random.RandomState, optional
-        The random state used for the shuffle.
+    random_state: int or numpy.random.RandomState, optional
+        Random seed used to initialize the numpy.random.RandomState number generator.
+        Excepts an instance of numpy.random.RandomState for backwards compatibility.
```
Comment (maciejkula) [inline lightfm/cross_validation.py]: Excepts => Accepts

**Comment by maciejkula** [comment]
Thank you!
